test/integration/consul_test.go

```diff
@@ -0,0 +1,106 @@
+//go:build integration
+// +build integration
+
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/nri-consul/src/args"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	containerIntegration = "nri-consul"
+	schema               = "consul-schema.json"
+)
+
+var (
+	containersConsul = []string{"consul-server1", "consul-server2", "consul-server3"}
+	responseRegex    = regexp.MustCompile(`{"name":"com\.newrelic\.consul".*`)
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	result := m.Run()
+	os.Exit(result)
+}
+
+func TestMetricsCollectionInCluterWithSSL(t *testing.T) {
+	if !waitForConsulClusterUpAndRunning(20) {
+		t.Fatal("tests cannot be executed")
+	}
+	defer dockerComposeDown()
```

Comment:
I think this is slightly different from integrations tests from others integrations where the make target runs the docker-compose up once. 
Why using this approach ? just to know the reason we break consistency

Reply:
https://github.com/newrelic/nri-nginx/blob/master/tests/integration/nginx_test.go

Reply:
I think there is a bit less boiler plate on that one

Reply:
I based myself in the one for [RabbitMQ](https://github.com/newrelic/nri-rabbitmq/tree/master/tests) and corrected it, because that one is not working, and I also added the docker-compose down in the test.
It's more correct, because we are not using a container running forever with a time.sleep instead we just run it once. Also, in this integration test, we are doing the docker-compose down in the test instead of having to add it in the make file.
I would actually correct the other integration tests to use a similar approach.

Reply:
And I actually thing there is less boilerplate because it's all in the helper, instead of having to add the docker-copmose and having a separate JsconSchemaValidator file.